### PR TITLE
[OSDOCS-4148] adds vSphere to supported k8s-nmstate platforms list

### PR DESCRIPTION
**Pull request by ShaunaDiaz**
This is the docs portions of [OSDOCS-4148](https://issues.redhat.com//browse/OSDOCS-4148)], adding vSphere to supported k8s-nmstate platforms list. No other docs are needed at this time.

Version(s):
  * PR applies to the in-development version and future versions: 4.12+

Link to docs preview:
https://51746--docspreview.netlify.app/openshift-enterprise/latest/networking/k8s_nmstate/k8s-nmstate-about-the-k8s-nmstate-operator.html

Additional information:
Release note: https://github.com/openshift/openshift-docs/pull/51249

### CODE REVIEW

**Comment by ShaunaDiaz** [comment]
@creydr Please let me know if looks good to you (or suggest changes).

**Review by creydr — APPROVED** [review]
/lgtm

**Review by tmalove — COMMENTED** [review]
/remove-label peer-review-needed
/label peer-review-in-progress

File: networking/k8s_nmstate/k8s-nmstate-about-the-k8s-nmstate-operator.adoc [hunk 1]
```diff
@@ -11,7 +11,7 @@ The Kubernetes NMState Operator provides a Kubernetes API for performing state-d
 
 [IMPORTANT]
 ====
-Red Hat supports the Kubernetes NMState Operator in production environments on bare-metal, IBM Power, IBM Z, and LinuxONE installations.
+Red Hat supports the Kubernetes NMState Operator in production environments on bare-metal, IBM Power, IBM Z, LinuxONE, and VMWare vSphere installations.
```
Comment (tmalove) [inline networking/k8s_nmstate/k8s-nmstate-about-the-k8s-nmstate-operator.adoc:14]:
```suggestion
Red Hat supports the Kubernetes NMState Operator in production environments on bare metal, IBM Power, IBM Z, LinuxONE, and VMWare vSphere installations.
```

**Review by tmalove — COMMENTED** [review]
/remove-label peer-review-in-progress
/label peer-review-done

File: networking/k8s_nmstate/k8s-nmstate-about-the-k8s-nmstate-operator.adoc (hunk 1 above)
Comment (ShaunaDiaz) [reply networking/k8s_nmstate/k8s-nmstate-about-the-k8s-nmstate-operator.adoc:14]: @tmalove In this case, bare-metal is a compound adjective: "bare-metal installations," so the hyphen is correct per the IBM Style Guide.

**Review by maxwelldb — APPROVED** [review]

**Pull request merged**